include/core/jtrace.h

```diff
@@ -58,13 +58,18 @@ void j_trace_leave (JTrace*);
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(JTrace, j_trace_leave)
 
+#ifndef JULEA_DEBUG
 #ifdef __COUNTER__
 #define J_TRACE(name, ...) g_autoptr(JTrace) G_PASTE(j_trace, __COUNTER__) G_GNUC_UNUSED = j_trace_enter(name, __VA_ARGS__)
 #define J_TRACE_FUNCTION(...) g_autoptr(JTrace) G_PASTE(j_trace_function, __COUNTER__) G_GNUC_UNUSED = j_trace_enter(G_STRFUNC, __VA_ARGS__)
 #else
 #define J_TRACE(name, ...) g_autoptr(JTrace) G_PASTE(j_trace, __LINE__) G_GNUC_UNUSED = j_trace_enter(name, __VA_ARGS__)
 #define J_TRACE_FUNCTION(...) g_autoptr(JTrace) G_PASTE(j_trace_function, __LINE__) G_GNUC_UNUSED = j_trace_enter(G_STRFUNC, __VA_ARGS__)
 #endif
+#else
+#define J_TRACE(name, ...) guint G_PASTE(j_trace, __LINE__) G_GNUC_UNUSED = 0;
+#define J_TRACE_FUNCTION(...) guint G_PASTE(j_trace_function, __LINE__) G_GNUC_UNUSED = 0;
+#endif
```

Comment:
This is missing the `__COUNTER__` case. It might make sense to introduce a new helper variable to avoid having to do the same checks twice.

Reply:
In the release build, tracing is completely removed. There is no need to make a difference between __COUNTER__ being set or not